pkg/operator/podsecurityreadinesscontroller/classification.go

```diff
@@ -0,0 +1,133 @@
+	if runLevelZeroNamespaces.Has(ns.Name) {
+		conditions.addViolatingRunLevelZero(ns)
+		return nil
+	}
+	if strings.HasPrefix(ns.Name, "openshift") {
+		conditions.addViolatingOpenShift(ns)
+		return nil
+	}
+	if ns.Labels[labelSyncControlLabel] == "false" {
+		conditions.addViolatingDisabledSyncer(ns)
+		return nil
+	}
+
+	isUserViolation, err := c.isUserViolation(ctx, ns, enforceLevel)
+	if err != nil {
+		klog.V(2).ErrorS(err, "Error checking user violations", "namespace", ns.Name)
+		// Transient API server error or temporary resource unavailability (most likely).
+		// Theoretically, psapi parsing errors could occur that retry without hope for recovery.
+		return err
+	}
+
+	if isUserViolation {
+		conditions.addViolatingUserSCC(ns)
+		return nil
+	}
+
+	// Historically, we assume that this is a customer issue, but
+	// actually it means we don't know what the root cause is.
+	conditions.addViolatingCustomer(ns)
+
+	return nil
+}
+
+func (c *PodSecurityReadinessController) isUserViolation(ctx context.Context, ns *corev1.Namespace, label string) (bool, error) {
+	var enforcementLevel psapi.Level
+	switch strings.ToLower(label) {
+	case "restricted":
+		enforcementLevel = psapi.LevelRestricted
+	case "baseline":
+		enforcementLevel = psapi.LevelBaseline
+	case "privileged":
+		// If privileged is violating, something is seriously wrong
+		// but testing against privileged level is pointless (everything passes)
+		klog.V(2).InfoS("Namespace violating privileged level - skipping user check",
+			"namespace", ns.Name)
+		return false, nil
+	default:
+		return false, fmt.Errorf("unknown level: %q", label)
+	}
+
+	allPods, err := c.kubeClient.CoreV1().Pods(ns.Name).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		klog.V(2).ErrorS(err, "Failed to list pods in namespace", "namespace", ns.Name)
+		return false, err
+	}
+
+	var userPods []corev1.Pod
+	for _, pod := range allPods.Items {
+		if strings.HasPrefix(pod.Annotations[securityv1.ValidatedSCCAnnotation], "restricted-v") {
+			// restricted-v2 is allowed for all system:authenticated, also for ServiceAccounts.
+			// But ServiceAccounts are not part of the group. So restricted-v2 will always
+			// result in user-based SCC. So we skip them as the user-based SCCs cause harm
+			// if they need a higher privileged than restricted.
+			// We watch for any restricted version above the first one. We might introduce
+			// restricted-v3 for user namespaces.
+			continue
+		}
+
+		if pod.Annotations[securityv1.ValidatedSCCSubjectTypeAnnotation] == "user" {
+			userPods = append(userPods, pod)
+		}
+	}
+
+	if len(userPods) == 0 {
+		return false, nil // No user pods = violation is from service accounts
+	}
+
+	enforcementVersion := psapi.LatestVersion()
+	for _, pod := range userPods {
+		klog.InfoS("Evaluating user pod against PSA level",
```

Comment:
Earlier in this file, we're logging at the `klog.V(2)` level, but not down in this section. Should that be consistent throughout (or perhaps even higher for the infos)?